libs/vm-modules/tests/unit/math/tensor_tests.cpp

```diff
@@ -135,42 +111,68 @@ TEST_F(MathTensorTests, tensor_4_dim_fixed64_fill)
   ASSERT_TRUE(toolkit.Run());
 }
 
-// Disabled until ML-329 resolved
-TEST_F(MathTensorTests, DISABLED_tensor_at_on_invalid_index)
+TEST_F(MathTensorTests, tensor_at_on_invalid_index)
 {
   static char const *SRC = R"(
-    function main() : Tensor
+    function main()
       var tensor_shape = Array<UInt64>(1);
       tensor_shape[0] = 2u64;
 
       var x = Tensor(tensor_shape);
-      var y = Tensor(tensor_shape);
-      x.fill(2.0fp64);
 
-      printLn(toString(x.at(999u64)));
+      printLn(toString(x.at(3u64)));
+    endfunction
+  )";
 
-     return y;
+  ASSERT_TRUE(toolkit.Compile(SRC));
+  EXPECT_FALSE(toolkit.Run());
+}
+
+TEST_F(MathTensorTests, tensor_at_invalid_index_count_too_many)
+{
+  static char const *SRC = R"(
+    function main()
+      var tensor_shape = Array<UInt64>(1);
+      tensor_shape[0] = 2u64;
+
+      var x = Tensor(tensor_shape);
+
+      printLn(toString(x.at(0u64, 0u64)));
     endfunction
   )";
 
   ASSERT_TRUE(toolkit.Compile(SRC));
   EXPECT_FALSE(toolkit.Run());
 }
 
-TEST_F(MathTensorTests, tensor_set_on_invalid_index)
+TEST_F(MathTensorTests, tensor_at_invalid_index_count_too_few)
 {
   static char const *SRC = R"(
-    function main() : Tensor
+    function main()
       var tensor_shape = Array<UInt64>(1);
       tensor_shape[0] = 2u64;
+      tensor_shape[1] = 2u64;
```

Comment:
i don't think this really tests anything about tensor though, you're just setting up an Array with one element and then trying to access its second element. That fails before you even try to construct a tensor

Reply:
Oh, no, my fault here: Array<UInt64>(2) was intended; 2-dim tensor with 1-dim accessor called.